codalab/objects/work_manager.py

```diff
@@ -175,6 +175,25 @@ def check_bundle_actions(self):
 
         return did_something
 
+    def check_timed_out_bundles(self, update_timeout=60*60*24):
+        '''
+        Filters the list of running bundles by their time, and marks those that
+        have not been updated in > update_timeout seconds as FAILED.
+        Default update_timeout:   1 day
+        '''
+        uuids = self.model.search_bundle_uuids(worksheet_uuid=None, user_id=self.model.root_user_id, \
+            keywords=['state='+','.join([State.RUNNING, State.QUEUED])])
+        bundles = self.model.batch_get_bundles(uuid=uuids)
+        def _failed(bundle):
+            now = int(time.time())
+            since_last_update = now - bundle.metadata.last_updated
+            return since_last_update >= update_timeout
+        failed_bundles = filter(_failed, bundles)
+        for bundle in failed_bundles:
+            metadata_update = {'failure_message': 'Job timed out'}
```

Comment:
Make the message more precise: "No response from worker in " + formatting.format_duration(update_timeout)